packages/domain/lib/usecase/get_cast_usecase.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:domain/model/cast/cast_model.dart';
+import 'package:yaml/yaml.dart';
+import '../model/cast/tmdb_image.dart';
+import '../model/cast/cast.dart';
+import '../model/cast/cast_with_images.dart';
+import '../repository/tmdb_api_repository.dart';
+import '../repository/trakt_api_repository.dart';
+import '../utils/getSecrets.dart';
+import 'usecase.dart';
+
+class GetCastUseCase
+    implements UseCaseParams<int?, Future<List<CastAndImages>?>> {
+  final TraktAPIRepository _traktAPIRepository;
+  final TmdbAPIRepository _tmdbAPIRepository;
+  String apiKey = '';
+
+  GetCastUseCase(
+    this._traktAPIRepository,
+    this._tmdbAPIRepository,
+  ) {
+    getAsyncApiKey();
+  }
```

Comment:
GetCastUseCase(
    this._traktAPIRepository,
    this._tmdbAPIRepository,
{this.apiKey = await getSecrets()["TMDB_API_KEY"]}
  );

Reply:
I can't use await not in the body of constructor. And also I don't want my constructor to be async